### PR TITLE
MSI auth support for NuGetGallery

### PR DESCRIPTION
**Pull request by agr**
Progress on https://github.com/NuGet/Engineering/issues/5335.

Added configuration options for authenticating with MSI for Gallery. Using storage account keys or SAS tokens for authentication still works.
Debug builds when `AzureStorageUseMsi == true` use `DefaultAzureCredential` instead to support local debugging.
Removed the reference to `WindowsAzure.Storage` package. Cleaned up tests.
Added blob properties extraction from `BlobDownloadDetails`, so calling `OpenReadAsync` would have blob properties prepared for consumption.

### CODE REVIEW

**Comment by lyndaidaii** [comment]
Looks like some conflicts, otherwise looks good to me.

**Comment by agr** [comment]
@lyndaidaii resolved conflicts.

File: src/NuGetGallery.Core/Services/CloudBlobClientWrapper.cs [hunk 1]
```diff
@@ -72,6 +72,20 @@ private CloudBlobClientWrapper(string storageConnectionString)
             return new CloudBlobClientWrapper(storageConnectionString, tokenCredential, readAccessGeoRedundant, requestTimeout);
         }
 
+        public static CloudBlobClientWrapper UsingDefaultAzureCredential(
+            string storageConnectionString,
+            string clientId = null,
+            bool readAccessGeoRedundant = false,
+            TimeSpan? requestTimeout = null)
+        {
+            var options = new DefaultAzureCredentialOptions
```
Comment (dannyjdev) [inline src/NuGetGallery.Core/Services/CloudBlobClientWrapper.cs:81]: I don't see this `options` variable being used, it should be pass in the `DefaultAzureCredential` constructor right?

Comment (agr) [reply src/NuGetGallery.Core/Services/CloudBlobClientWrapper.cs:81]: Fixed!

**Review by drewgillies — APPROVED** [review]

**Comment by drewgillies** [comment]
Discussed offline.

**Pull request merged**